frontend/src/components/ProfileCard.jsx

```diff
@@ -0,0 +1,263 @@
+    };
+
+    const handleCancel = () => {
+      setEditedUser(currentUser);
+      setIsEditMode(false);
+    };
+
+
+    return (
+      <>
+        <Button variant='contained' color='success' disableElevation onClick={handleSave}>Save</Button>
+        <Button style={{ marginLeft: '4px' }} variant='contained' color='error' disableElevation onClick={handleCancel}>Cancel</Button>
+      </>
+    );
+  };
+
+  const usernameField = (
+    <Box fontSize={'12'} sx={{ paddingBottom: '10px', display: 'flex', alignItems: 'flex-end' }}>
+      <Box sx={{ display: 'flex', alignItems: 'flex-end' }}>
+        <Box paddingBottom='1px'>
+          <Typography variant='h6' style={{ color: 'action.active' }}>@</Typography>
+        </Box>
+        <TextField
+          size='small'
+          label={isEditMode ? 'Username' : ''}
+          variant='standard'
+          onChange={e => setEditedUser({ ...editedUser, username: e.target.value })}
+          value={editedUser.username}
+          disabled={!isEditMode}
+          InputProps={{ disableUnderline: !isEditMode }}
+          inputProps={{ style: { fontSize: '18px' } }}
+          sx={{
+            '& .MuiInputBase-input.Mui-disabled': {
+              WebkitTextFillColor: '#000000',
+            },
+          }} 
+        />
+      </Box>
+    </Box>
+  );
+
+  const imgInputStyle = {
+    clip: 'rect(0 0 0 0)',
+    clipPath: 'inset(50%)',
+    height: 1,
+    overflow: 'hidden',
+    position: 'absolute',
+    bottom: 0,
+    left: 0,
+    whiteSpace: 'nowrap',
+    width: 1,
+  };
+
+  return (
+    <>
+      <Box>
+        <Grid container spacing={2}>
+          <Grid xs={3} item alignSelf={'center'}>
+            <Tooltip title={isEditMode ? 'Change Profile Picture' : ''}>
+              <IconButton 
+                component='label'
+                disabled={!isEditMode}
+              >
+                <CardMedia
+                  component='img'
+                  image='/Snoopy-doghouse-1.jpg'
+                  alt='Snoopy doghouse'
+                  sx={{ padding: '4px',  borderRadius: '50%', border: `${isEditMode ? '1px dashed grey' : ''}` }}
+                  style={{ width: '100%', height: 'auto' }}
+                />
+                {/* TODO upload an actual image */}
+                <Input type='file' style={imgInputStyle} />
+              </IconButton>
+            </Tooltip>
+          </Grid>
+          <Grid item xs={9}>
+            <CardHeader 
+              action={(isAdmin || isSelf) && (isEditMode ? saveProfileButton() : editProfileButton())}
+              title={usernameField}
+              subheader={isAdmin ? `User ID: ${_id}` : ''}
+              subheaderTypographyProps={{
+                variant: 'subtitle2',
+                color: 'gret',
+              
+              }}
+            />
+            <CardContent sx={{ flex: '2 0 auto' }}>
+              <div style={{ margin: '10px' }}>
+                <InputLabel sx={{ color: 'black' }}>{'Name: '}
```

Comment:
nit i don't think we should expose this on the frontend since it is for db use only